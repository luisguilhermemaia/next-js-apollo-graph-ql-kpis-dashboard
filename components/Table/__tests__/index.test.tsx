import '@testing-library/jest-dom/extend-expect';
import { RenderResult, render } from '@testing-library/react';
import React from 'react';
import { Table } from '../index';

describe('Table', () => {
  let renderResult: RenderResult;
  const data = [
    {
      id: '#6b3b2a',
      name: 'Hermiston and Sons',
      segment: 'iterate enterprise e-markets',
      contract: 'Kling - Herman',
      renewals: 35331,
      isCritical: false,
      nps: {
        id: '5169aed7-9f4e-4fad-8c2c-6db25d917ce1',
        currentPeriod: 93487,
        lastPeriod: 98934,
        avg: 8574,
      },
    },
    {
      id: '#75604c',
      name: 'Stiedemann - Mraz',
      segment: 'benchmark dot-com metrics',
      contract: 'Kutch - Schmeler',
      renewals: 9300,
      isCritical: false,
      nps: {
        id: '38ce90ad-eecc-469e-a06e-dd2d135886cb',
        currentPeriod: 74749,
        lastPeriod: 46816,
        avg: 65374,
      },
    },
    {
      id: '#5d0a67',
      name: 'Williamson Group',
      segment: 'recontextualize wireless solutions',
      contract: "O'Conner, Schneider and Senger",
      renewals: 14901,
      isCritical: false,
      nps: {
        id: '8a4f5c49-1ffe-48f8-ad30-09435605bc6d',
        currentPeriod: 67119,
        lastPeriod: 22756,
        avg: 96744,
      },
    },
    {
      id: '#404524',
      name: 'Koelpin - Veum',
      segment: 'exploit distributed metrics',
      contract: 'Schuppe, Champlin and Jast',
      renewals: 99965,
      isCritical: true,
      nps: {
        id: 'b71608fb-b1ea-46d3-9482-1b448e2c2c85',
        currentPeriod: 74465,
        lastPeriod: 79164,
        avg: 20045,
      },
    },
    {
      id: '#2e1b58',
      name: 'Rice Group',
      segment: 'exploit cutting-edge e-commerce',
      contract: 'Dooley and Sons',
      renewals: 34362,
      isCritical: true,
      nps: {
        id: '6fc8adcc-cc0b-4941-8b27-767847bf73f5',
        currentPeriod: 15497,
        lastPeriod: 45342,
        avg: 94705,
      },
    },
    {
      id: '#296f19',
      name: 'Legros, Morissette and Farrell',
      segment: 'extend proactive experiences',
      contract: 'Kemmer, Schneider and Zieme',
      renewals: 36928,
      isCritical: true,
      nps: {
        id: '5466ee83-45a2-4136-bacd-fb2a005e987d',
        currentPeriod: 81125,
        lastPeriod: 95259,
        avg: 14959,
      },
    },
    {
      id: '#15791b',
      name: 'Olson and Sons',
      segment: 'leverage proactive platforms',
      contract: 'Barton Group',
      renewals: 91614,
      isCritical: false,
      nps: {
        id: '24253d70-0f21-4f03-9cab-9909adb4ddbd',
        currentPeriod: 61852,
        lastPeriod: 42321,
        avg: 92146,
      },
    },
    {
      id: '#571545',
      name: 'Koch LLC',
      segment: 'extend clicks-and-mortar communities',
      contract: 'Keebler, Pfannerstill and Lakin',
      renewals: 81438,
      isCritical: false,
      nps: {
        id: '62f19572-66a6-408a-aab3-35b52998e5ee',
        currentPeriod: 63790,
        lastPeriod: 98423,
        avg: 79988,
      },
    },
    {
      id: '#4b0a50',
      name: 'Ryan - Kshlerin',
      segment: 'strategize granular methodologies',
      contract: 'Bode and Sons',
      renewals: 43895,
      isCritical: false,
      nps: {
        id: '41ff0b83-e074-4ce9-910c-527166131650',
        currentPeriod: 51730,
        lastPeriod: 54266,
        avg: 34529,
      },
    },
    {
      id: '#51331f',
      name: 'Schmeler - Thiel',
      segment: 'incentivize robust initiatives',
      contract: 'Dickinson, Wehner and Rolfson',
      renewals: 71148,
      isCritical: true,
      nps: {
        id: 'b02f391b-4af1-445b-8960-94e62fbbbc80',
        currentPeriod: 31684,
        lastPeriod: 37333,
        avg: 16918,
      },
    },
  ];

  const sortState = { value: 'id', label: 'ID' };

  const doRender = (showCritical: boolean) => {
    renderResult = render(
      <Table
        loading={false}
        companies={data}
        sortState={sortState}
        showCritical={showCritical}
      />
    );
  };

  it('renders correctly with showCritical false', () => {
    doRender(false);

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with showCritical true', () => {
    doRender(true);

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });
});
