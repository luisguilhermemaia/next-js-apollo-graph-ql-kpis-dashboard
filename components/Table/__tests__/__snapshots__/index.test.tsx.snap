// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly with showCritical false 1`] = `
<div>
  <table
    class="Table__StyledTable-ch5b9r-3 dVhPvg"
  >
    <thead
      class="Table__StyledTableHead-ch5b9r-2 hejyOx"
    >
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Id
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Company name
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Segment
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Contract
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          RENEWALS
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          NPS avg
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          NPS last
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          NPS First
        </th>
      </tr>
    </thead>
    <tbody
      class="Table__StyledTableBody-ch5b9r-0"
    >
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 gWeppk"
            color="#15791b"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Olson and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          leverage proactive platforms
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Barton Group
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          91614
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          92146
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          42321
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          61852
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 isShem"
            color="#296f19"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Legros, Morissette and Farrell
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          extend proactive experiences
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Kemmer, Schneider and Zieme
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          36928
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          14959
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          95259
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          81125
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 mQTzs"
            color="#2e1b58"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Rice Group
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          exploit cutting-edge e-commerce
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Dooley and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          34362
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          94705
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          45342
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          15497
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 bKTsEe"
            color="#404524"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Koelpin - Veum
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          exploit distributed metrics
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Schuppe, Champlin and Jast
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          99965
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          20045
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          79164
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          74465
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 eVOcxr"
            color="#4b0a50"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Ryan - Kshlerin
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          strategize granular methodologies
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Bode and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          43895
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          34529
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          54266
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          51730
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 jJTLyC"
            color="#51331f"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Schmeler - Thiel
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          incentivize robust initiatives
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Dickinson, Wehner and Rolfson
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          71148
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          16918
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          37333
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          31684
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 dbEqIq"
            color="#571545"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Koch LLC
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          extend clicks-and-mortar communities
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Keebler, Pfannerstill and Lakin
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          81438
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          79988
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          98423
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          63790
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 dAoawM"
            color="#5d0a67"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Williamson Group
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          recontextualize wireless solutions
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          O'Conner, Schneider and Senger
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          14901
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          96744
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          22756
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          67119
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 bYltkr"
            color="#6b3b2a"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Hermiston and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          iterate enterprise e-markets
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Kling - Herman
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          35331
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          8574
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          98934
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          93487
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 kHXntG"
            color="#75604c"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Stiedemann - Mraz
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          benchmark dot-com metrics
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Kutch - Schmeler
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          9300
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          65374
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          46816
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          74749
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table renders correctly with showCritical true 1`] = `
<div>
  <table
    class="Table__StyledTable-ch5b9r-3 dVhPvg"
  >
    <thead
      class="Table__StyledTableHead-ch5b9r-2 hejyOx"
    >
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Id
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Company name
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Segment
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          Contract
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          RENEWALS
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          NPS avg
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          NPS last
        </th>
        <th
          class="Table__StyledTableHeader-ch5b9r-5 foRQsW"
        >
          NPS First
        </th>
      </tr>
    </thead>
    <tbody
      class="Table__StyledTableBody-ch5b9r-0"
    >
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 gWeppk"
            color="#15791b"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Olson and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          leverage proactive platforms
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Barton Group
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          91614
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          92146
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          42321
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          61852
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 bmanpx"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 isShem"
            color="#296f19"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Legros, Morissette and Farrell
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          extend proactive experiences
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Kemmer, Schneider and Zieme
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          36928
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          14959
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          95259
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          81125
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 bmanpx"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 mQTzs"
            color="#2e1b58"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Rice Group
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          exploit cutting-edge e-commerce
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Dooley and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          34362
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          94705
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          45342
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          15497
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 bmanpx"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 bKTsEe"
            color="#404524"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Koelpin - Veum
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          exploit distributed metrics
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Schuppe, Champlin and Jast
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          99965
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          20045
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          79164
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          74465
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 eVOcxr"
            color="#4b0a50"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Ryan - Kshlerin
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          strategize granular methodologies
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Bode and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          43895
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          34529
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          54266
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          51730
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 bmanpx"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 jJTLyC"
            color="#51331f"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Schmeler - Thiel
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          incentivize robust initiatives
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Dickinson, Wehner and Rolfson
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          71148
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          16918
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          37333
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          31684
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 dbEqIq"
            color="#571545"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Koch LLC
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          extend clicks-and-mortar communities
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Keebler, Pfannerstill and Lakin
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          81438
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          79988
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          98423
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          63790
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 dAoawM"
            color="#5d0a67"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Williamson Group
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          recontextualize wireless solutions
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          O'Conner, Schneider and Senger
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          14901
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          96744
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          22756
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          67119
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 bYltkr"
            color="#6b3b2a"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Hermiston and Sons
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          iterate enterprise e-markets
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Kling - Herman
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          35331
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          8574
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          98934
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          93487
        </td>
      </tr>
      <tr
        class="Table__StyledTableRow-ch5b9r-4 cFbMMw"
      >
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Id"
        >
          <span
            class="Table__StyledDot-ch5b9r-1 kHXntG"
            color="#75604c"
          />
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Company name"
        >
          Stiedemann - Mraz
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Segment"
        >
          benchmark dot-com metrics
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="Contract"
        >
          Kutch - Schmeler
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="RENEWALS"
        >
          9300
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS avg"
        >
          65374
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS last"
        >
          46816
        </td>
        <td
          class="Table__StyledTableCell-ch5b9r-6 euguko"
          data-label="NPS First"
        >
          74749
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
